src/main/java/bio/terra/service/snapshot/flight/delete/SnapshotDeleteFlight.java

```diff
@@ -33,15 +42,35 @@ public SnapshotDeleteFlight(FlightMap inputParameters, Object applicationContext
         BigQueryPdao bigQueryPdao = (BigQueryPdao)appContext.getBean("bigQueryPdao");
         ResourceService resourceService = (ResourceService) appContext.getBean("resourceService");
         IamService iamClient = (IamService)appContext.getBean("iamService");
+        DatasetDao datasetDao = (DatasetDao) appContext.getBean("datasetDao");
```

Comment:
We could use `appContext.getBean("datasetDao", DatasetDao.class)` to remove the need for casting here and in the other invocations of `getBean`. From the Spring docs:
```
Return an instance, which may be shared or independent, of the specified bean.
	 <p>Behaves the same as {@link #getBean(String)}, but provides a measure of type
	 safety by throwing a BeanNotOfRequiredTypeException if the bean is not of the
	 required type. This means that ClassCastException can't be thrown on casting
	 the result correctly, as can happen with {@link #getBean(String)}.```
	 
Not really a necessary change, so I think it could be merged without it. Just something to keep in mind.

Reply:
That's good to know! I think we can make this a small follow-up ticket since it looks like a lot of places are using casting.